### PR TITLE
coneccting the mail system to the front-end

### PR DESCRIPTION
**Pull request by talalalamdar**
coneccting the mail system to the front-end:

  1- modifying sendAnEmail() function in sendGrid.js, and adding a route for it in routes.js

  2- adding a new sendAnEmail() function in src/util/index.js so it can be used in the front end

### CODE REVIEW

File: server/api/sendGrid.js [hunk 1]
```diff
@@ -5,15 +5,21 @@ const config = require("../config/config.js")
...
 
-function sendAnEmail(recipient, sender, subject, text) {
+function sendAnEmail(req, res) {
     const msg = {
-        to: recipient,
-        from: sender,
-        subject: subject,
-        text: text,
-        html: `<strong>${text}</strong>`,
+        to: req.body.recipient,
```
Comment (JawharBairakdar) [inline server/api/sendGrid.js]: - [ ] what if the sender needs to send for more than one recipient?

File: server/api/sendGrid.js [hunk 2]
```diff
@@ -5,15 +5,21 @@ const config = require("../config/config.js")
...
+        subject: req.body.subject,
+        text: req.body.text,
+        html: `<strong>${req.body.text}</strong>`,
     }
-    sgMail.send(msg)
+    sgMail
+        .send(msg)
+        .then(() => {
+            console.log("mail sent successfully")
+            res.end()
```
Comment (JawharBairakdar) [inline server/api/sendGrid.js]: - [ ] can we send a message to the client side telling them that the email has been sent with the correct status for it?

File: server/api/sendGrid.js (hunk 1 above)
Comment (malnajar) [reply server/api/sendGrid.js]: you can do that from the front end but it is better if he fix it

Comment (talalalamdar) [reply server/api/sendGrid.js]: you can actually, just simply add the first parameter as an array of strings

File: server/api/sendGrid.js (hunk 2 above)
Comment (talalalamdar) [reply server/api/sendGrid.js]: @JawharBairakdar yeah you can do that but not from here, you will need to modify the new function  sendAnEmail() in src/util/index.js (front-end), depends on what we need like set a state for a popup window when the email is sent

Comment (JawharBairakdar) [reply server/api/sendGrid.js]: Actully all of what we need is only the state and a message that says the email has been sent.
you can do it from a multible places it's your code you can chose what you prefer 🙂.

I am Actully wondering if you can send a status code if the email has been faild for a reason so the sender can know that it didn't arrive.

**Comment by JawharBairakdar** [comment]
Okay then nice work two small fixes to merge :)

**Comment by talalalamdar** [comment]
@JawharBairakdar yeah you are right but i still dont know where are we putting the send email functionality  button, but where ever it's gonna be we can just add a state for that component and act depends on that, do you know where are we in the UI sending an email ??

**Comment by JawharBairakdar** [comment]
You do't have to worry about the button that will send you the Information that you want from it because you can controle what you want from it so you are the peson who should decided all of the functionality of it.

**Pull request merged**